### PR TITLE
Unset variables

### LINKED ISSUE #2: API not ideal when exclusively unsetting environment variables
I noticed that the API is a bit clumsy, when only unsetting variables.

The following does not compile:
```rust
#[test]
    fn test_function_works_when_env_var_missing() {
        temp_env::with_var("SHOULD_BE_MISSING", None, || {
            let actual = env::var("SHOULD_BE_MISSING");
            let expected = Result::Err(VarError::NotPresent);
            assert_eq!(expected, actual);
        })
    }
```

It only works like so:
```rust
#[test]
    fn test_function_works_when_env_var_missing() {
        //                vvvvvvvvvvvv
        temp_env::with_var::<_,&str,_>("SHOULD_BE_MISSING", None, || {
            let actual = env::var("SHOULD_BE_MISSING");
            let expected = Result::Err(VarError::NotPresent);
            assert_eq!(expected, actual);
        })
    }
```

The compiler error is not that nice either:
```
error[E0283]: type annotations needed
   --> src/test.rs:111:9
    |
111 |         temp_env::with_var("SHOULD_BE_MISSING", None, || {
    |         ^^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `V` declared on the function `with_var`
    |
    = note: cannot satisfy `_: AsRef<OsStr>`
note: required by a bound in `with_var`
   --> /Users/fabian.braun/.cargo/registry/src/github.com-1ecc6299db9ec823/temp-env-0.1.0/src/lib.rs:57:8
    |
57  |     V: AsRef<OsStr> + Clone,
    |        ^^^^^^^^^^^^ required by this bound in `with_var`
help: consider specifying the type arguments in the function call
    |
111 |         temp_env::with_var::<K, V, F>("SHOULD_BE_MISSING", None, || {
    |                           +++++++++++
```

This is a side-effect of using `AsRef<OsStr>` as a type bound instead of a concrete type. I assume you made that change to comply with the API of `env::with_var`. Do you have an idea how to go about this?

### PR DESCRIPTION
**Pull request by fabian-braun**
closes #2

### CODE REVIEW

**Review by vmx — COMMENTED** [review]
Thanks, that's a good addition to the API. I left comments about some minor things, the rest looks good.

File: src/lib.rs [hunk 1]
```diff
@@ -143,7 +190,7 @@ mod tests {
         let foo_is_set = env::var("FOO").unwrap();
         assert_eq!(foo_is_set, "bar", "`FOO` must be set to \"bar\".");
 
-        crate::with_var("FOO", None::<&str>, || {
+        crate::with_var_unset("FOO", || {
```
[the remark this answers is not in the document]
Comment (fabian-braun) [reply src/lib.rs]: Yes, that makes sense. I added a dedicated test for the new function.

**Review by vmx — APPROVED** [review]
Thanks, this change looks good. I left one comment about the type annotation, which is just FYI, it doesn't prevent merging this pull request. So feel free to merge.

I personally prefer having a clean commit history, so I'd probably squash merge this commit. But again, I leave it up to you, it's fine either way.

**Comment by fabian-braun** [comment]
> Thanks, this change looks good. I left one comment about the type annotation, which is just FYI, it doesn't prevent merging this pull request. So feel free to merge.
> 
> I personally prefer having a clean commit history, so I'd probably squash merge this commit. But again, I leave it up to you, it's fine either way.

thanks! I did as you suggested. I was wondering if you have any recommendations on code-autoformatting setup. My IDE IntelliJ Ultimate is frequently screwing things up, apparently, but when I run `cargo fmt --all -- --check` locally it doesn't complain. In [the pipeline](https://github.com/vmx/temp-env/runs/4446834025?check_suite_focus=true) it does though, which confuses me a bit

**Comment by vmx** [comment]
I sadly don't have any recommendations. I usually run `cargo fmt` from command line and it does the right thing.

**Pull request merged**